FrozenPartitionTableTest.partitionTable_shouldBeFixed_whenMemberRestarts_usingNewUuid failure

**Error Message**

```
com.hazelcast.spi.exception.WrongTargetException: WrongTarget! local: Member [127.0.0.1]:5703 - bd5d6faf-9735-4300-8038-c4ac23167ce8 this, expected-target: null, partitionId: 0, replicaIndex: 0, operation: com.hazelcast.internal.partition.impl.FrozenPartitionTableTest$NonRetryablePartitionOperation, service: null
```
**Stacktrace**
```
java.util.concurrent.CompletionException: com.hazelcast.spi.exception.WrongTargetException: WrongTarget! local: Member [127.0.0.1]:5703 - bd5d6faf-9735-4300-8038-c4ac23167ce8 this, expected-target: null, partitionId: 0, replicaIndex: 0, operation: com.hazelcast.internal.partition.impl.FrozenPartitionTableTest$NonRetryablePartitionOperation, service: null
	at com.hazelcast.spi.impl.AbstractInvocationFuture.returnOrThrowWithJoinConventions(AbstractInvocationFuture.java:761)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.resolveAndThrowWithJoinConvention(AbstractInvocationFuture.java:777)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.join(AbstractInvocationFuture.java:537)
	at com.hazelcast.internal.partition.impl.FrozenPartitionTableTest.partitionTable_shouldBeFixed_whenMemberRestarts_usingNewUuid(FrozenPartitionTableTest.java:231)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
Caused by: com.hazelcast.spi.exception.WrongTargetException: WrongTarget! local: Member [127.0.0.1]:5703 - bd5d6faf-9735-4300-8038-c4ac23167ce8 this, expected-target: null, partitionId: 0, replicaIndex: 0, operation: com.hazelcast.internal.partition.impl.FrozenPartitionTableTest$NonRetryablePartitionOperation, service: null
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.ensureNoPartitionProblems(OperationRunnerImpl.java:375)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:231)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:452)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:166)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:136)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.executeRun(OperationThread.java:123)
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102)
```
http://jenkins.hazelcast.com/job/Hazelcast-pr-builder/2262/

--- a/hazelcast/src/test/java/com/hazelcast/internal/partition/impl/FrozenPartitionTableTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/internal/partition/impl/FrozenPartitionTableTest.java
@@ -223,9 +223,10 @@ public class FrozenPartitionTableTest extends HazelcastTestSupport {
         hz3.shutdown();
         assertClusterSizeEventually(2, hz1, hz2);
 
-        newHazelcastInstance(initOrCreateConfig(new Config()),
+        hz3 = newHazelcastInstance(initOrCreateConfig(new Config()),
                 randomName(), new StaticMemberNodeContext(factory, newUnsecureUUID(), member3.getAddress()));
         assertClusterSizeEventually(3, hz1, hz2);
+        waitAllForSafeState(hz1, hz2, hz3);
 
         OperationServiceImpl operationService = getOperationService(hz1);
         operationService.invokeOnPartition(null, new NonRetryablePartitionOperation(), member3PartitionId).join();

